runtime/runtime-params-estimator/src/rocksdb.rs

```diff
@@ -0,0 +1,311 @@
+const ANOTHER_PRANDOM_SEED: u64 = 0x0465b6733af62af0;
+const INPUT_DATA_BUFFER_SIZE: usize = (bytesize::MIB as usize) - 1;
+
+pub(crate) fn rocks_db_inserts_cost(config: &Config) -> GasCost {
+    let db_config = &config.rocksdb_test_config;
+    let data = input_data(db_config, INPUT_DATA_BUFFER_SIZE);
+    let tmp_dir = tempfile::TempDir::new().expect("Failed to create directory for temp DB");
+    let db = new_test_db(&tmp_dir, &data, &db_config);
+
+    if db_config.debug_rocksdb {
+        println!("# {:?}", db_config);
+        println!("# After setup / before measurement:");
+        print_levels_info(&db);
+    }
+
+    let gas_counter = GasCost::measure(config.metric);
+
+    if db_config.sequential_keys {
+        sequential_inserts(
+            db_config.op_count,
+            db_config.value_size,
+            &data,
+            db_config.setup_insertions,
+            &db,
+            db_config.force_compaction,
+            db_config.force_flush,
+        );
+    } else {
+        prandom_inserts(
+            db_config.op_count,
+            db_config.value_size,
+            &data,
+            ANOTHER_PRANDOM_SEED,
+            &db,
+            db_config.force_compaction,
+            db_config.force_flush,
+        );
+    }
+
+    let cost = gas_counter.elapsed();
+
+    if db_config.debug_rocksdb {
+        println!("# Cost: {:?}", cost);
+        print_levels_info(&db);
+    }
+
+    drop(db);
+    tmp_dir.close().expect("Could not clean up temp DB");
+
+    // Store generated input data in a file for reproducibility of any results
+    if db_config.pr_data_path.is_none() {
+        let mut stats_file = std::fs::OpenOptions::new()
+            .read(true)
+            .write(true)
+            .append(true)
+            .create(true)
+            .open("names-to-stats.txt")
+            .unwrap();
+        let stats_num = std::io::BufReader::new(&stats_file).lines().count();
+        let data_dump_path = format!("data_dump_{:<04}.bin", stats_num);
+
+        let mut dump = std::fs::File::create(&data_dump_path).unwrap();
+        dump.write_all(&data).unwrap();
+
+        writeln!(stats_file, "# DATA {} written to {}", stats_num, data_dump_path)
+            .expect("Writing to \"names-to-stats.txt\" failed");
+    }
+
+    cost
+}
+
+pub(crate) fn rocks_db_read_cost(config: &Config) -> GasCost {
+    let db_config = &config.rocksdb_test_config;
+    let tmp_dir = tempfile::TempDir::new().expect("Failed to create directory for temp DB");
+    let data = input_data(db_config, INPUT_DATA_BUFFER_SIZE);
+    let db = new_test_db(&tmp_dir, &data, &db_config);
+
+    if db_config.debug_rocksdb {
+        println!("# {:?}", db_config);
+        println!("# After setup / before measurement:");
+        print_levels_info(&db);
+    }
+
+    let mut prng: XorShiftRng = rand::SeedableRng::seed_from_u64(SETUP_PRANDOM_SEED);
+    let mut keys: Vec<usize> = (0..db_config.setup_insertions).map(|_| prng.gen()).collect();
```

Comment:
`iter::repat_with(|| prng.gen()).take(db_config.setup_insertions)` is another way to write this.

Reply:
Ah yes, that's a bit nicer here :)